org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/ProjectsManager.java

```diff
@@ -218,8 +221,9 @@ public void cleanupResources(IProject project) throws CoreException {
 	}
 
 	private void deleteInvalidProjects(Collection<IPath> rootPaths, IProgressMonitor monitor) {
+		List<String> workspaceProjects = rootPaths.stream().map((IPath rootPath) -> JDTStandaloneFileUtils.getWorkspaceInvisibleProjectName(rootPath)).collect(Collectors.toList());
 		for (IProject project : getWorkspaceRoot().getProjects()) {
-			if (project.exists() && (ResourceUtils.isContainedIn(project.getLocation(), rootPaths) || ProjectUtils.isGradleProject(project))) {
+			if (project.exists() && (ResourceUtils.isContainedIn(project.getLocation(), rootPaths) || ProjectUtils.isGradleProject(project)) || workspaceProjects.contains(project.getName())) {
```

Comment:
gradle check here

Reply:
The gradle check is added by another PR https://github.com/eclipse/eclipse.jdt.ls/commit/320401b60ec9caf036345fc3cd24d44211216fb4#diff-54b8440d3466866733ad3374655cf2bf, not sure the underlying reason.